google/api_core/bidi.py

```diff
@@ -276,7 +276,11 @@ def open(self):
         request_generator = _RequestQueueGenerator(
             self._request_queue, initial_request=self._initial_request
         )
-        call = self._start_rpc(iter(request_generator), metadata=self._rpc_metadata)
+        try:
+            call = self._start_rpc(iter(request_generator), metadata=self._rpc_metadata)
+        except exceptions.GoogleAPICallError as exc:
+            self._on_call_done(exc)
```

Comment:
The exception will be raised in the call back `add_done_callback` and must be handled by the user. This is based on the comment here:

https://github.com/googleapis/python-api-core/blob/8f73d2ee2d3af2201f877aa7e2f7361147759dc7/google/api_core/bidi.py#L608-L610

Reply:
`add_done_callback` [specifies](https://github.com/googleapis/python-api-core/blob/887751e114fd6ae8276529c0b9b726a0d5d7c8eb/google/api_core/bidi.py#L261) `callback (Callable[[grpc.Future], None])`, but it's not clear to me that `GoogleAPICallError` is a `grpc.Future` (see [this](https://googleapis.dev/python/google-api-core/latest/exceptions.html#google.api_core.exceptions.GoogleAPICallError))

Reply:
The change that I proposed is exactly the same behaviour as the `_on_call_done` method of `ResumableBidiRpc`

https://github.com/googleapis/python-api-core/blob/6251eab3fca5f7e509cb9b6e476ce1184094b711/google/api_core/bidi.py#L441-L443

From https://github.com/grpc/grpc/issues/10885#issuecomment-302651331, `grpc.RpcError` is also `grpc.Call`.

Reply:
I reverted https://github.com/googleapis/python-api-core/commit/c7f63bd270d81f75abdf0282bd3c4be4de02bdbd because `GoogleAPICallError` is not `grpc.Call` or `grpc.RpcError`. The original `grpc.RpcError` is available on the ``response`` property of  `GoogleAPICallError`

https://github.com/googleapis/python-api-core/blob/6251eab3fca5f7e509cb9b6e476ce1184094b711/google/api_core/grpc_helpers.py#L174-L182

The behaviour proposed in this PR is the same as `ResumableBidiRpc` and we typically raise `GoogleAPICallError` instead of `grpc.RpcError`

Reply:
I changed the code in [753a591](https://github.com/googleapis/python-api-core/pull/357/commits/753a591bb7f9603929fc81c6880c58237c23a49c) to raise `grpc.RpcError` instead of `GoogleAPICallError` since `grpc.RpcError` is also `grpc.Call` and it is a better fit